### PR TITLE
💎 Update tags at scope instead of updating VMs

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: This fixes a TODO in the AzureMachine controller to use the new tags' SDK CreateOrUpdateAtScope (released in v0.45.1). This allows:
1. Making the update more targeted, updating tags only instead of the entire VM resource
2. Unit testing the tags reconciler
3. in the future the same createOrUpdateAtScope tags reconciler could be used for other Azure resources

Fixed a few typos on the way.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Use tags CreateOrUpdateAtScope to ensure VM tags are correct after the VM is created
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
I tried running this code and all looks good 

also 👍 for all the cleanup

File: api/v1alpha3/tags.go [hunk 1]
```diff
@@ -118,6 +118,12 @@ const (
 
 	// CommonRole describes the value for the common role
 	CommonRole = "common"
+
+	// TagsLastAppliedAnnotation is the key for the machine object annotation
```
Comment (CecileRobertMichon) [inline api/v1alpha3/tags.go:122]: just noticed need to fix the comment to match the name here

Comment (nader-ziada) [reply api/v1alpha3/tags.go:122]: missed that one

**Pull request merged**